src/main/scala/edu/ie3/mobsim/model/ElectricVehicle.scala

```diff
@@ -144,10 +143,18 @@ final case class ElectricVehicle(
     copy(chargingAtSimona = false)
   }
 
-  def setFinalDestinationPoi(
-      destinationPoi: Option[PointOfInterest]
+  def resetFinalDestination(): ElectricVehicle = {
+    copy(finalDestinationPoi = None, finalDestinationPoiType = None)
+  }
+
+  def setFinalDestination(
+      destinationPoi: PointOfInterest,
+      destinationPoiType: PoiTypeDictionary.Value
```

Comment:
It doesn't get clear to me why the Option in no longer necessary. Maybe you could give me some explanation here?

Reply:
this is now split into two methods `setFinalDestination`and `resetFinalDestination`. Since we have to set the `poi` and `poiType` we always want to either set both of them with an actual value or reset both of them to None. With these method signatures we assure that we don't set one of them so `Some(x)` and the other one to None which we wouldn't want. Also it makes the method call more concise since you do not have to call them with optionals.